packages/calcite-components/src/components/popover/popover.tsx

```diff
@@ -292,7 +292,23 @@ export class Popover
     this.setFilteredPlacements();
     connectLocalized(this);
     connectMessages(this);
-    connectFocusTrap(this);
+    connectFocusTrap(this, {
+      focusTrapEl: this.el,
+      focusTrapOptions: {
+        allowOutsideClick: true,
+        clickOutsideDeactivates: (event: MouseEvent) => {
+          const isReferenceElementInPath =
+            this.referenceElement instanceof EventTarget &&
+            event.composedPath().includes(this.referenceElement);
+
+          const outsideClick = !event.composedPath().includes(this.el);
```

Comment:
Can you DRY this up by storing `event.composedPath`, and `autoClose && outsideClick` in a local variable?